### PR TITLE
Fixes to `fast-cmdline`

### PR DESCRIPTION
**Pull request by danilexn**
This proposes some fixes and changes to the current `fast-cmdline` branch from Marvin:

- Merged changes from the `merging_issue` branch, e.g., It was not generating the barcode counts for puck collection generation
- Handling the validation of `map_strategy` when added/updated through the cmdline (issue #54)
- Added some warnings

Ongoing:

- [ ] Checking all unit tests from Marvin...
- [ ] ... and implementing unit tests for the _tiny_ spatial data, and some more
- [ ] Do not generate puck_collection when there is only one puck

To discuss:

- When reverting a current project_df to previous version (e.g., current one in master #[55f10e2](https://github.com/rajewsky-lab/spacemake/commit/55f10e21c313d20350c2171f78c1bb26e35636f9)), it does not work anymore because of the changes in the map strategy format. We need to document this better

### CODE REVIEW

File: spacemake/cmdline.py [hunk 1]
```diff
@@ -824,7 +842,6 @@ def spacemake_run(args):
         keepgoing=args["keep_going"],
         printshellcmds=args["printshellcmds"],
         config=config_variables,
-        rerun_triggers=["mtime"],
```
Comment (marvin-jens) [inline spacemake/cmdline.py:827]: why remove this? I added this line to restore the old snakemake behaviour. Current versions rerun even if you change source files or rules files, which is excessive.

**Review by marvin-jens — CHANGES_REQUESTED** [review]
looks fine to me! We need testing of the puck collection/merging code though. The only question I have is regarding the `rerun_trigger=['mtime']` removal.

File: spacemake/cmdline.py (hunk 1 above)
Comment (danilexn) [reply spacemake/cmdline.py:827]: You're right, I removed this by accident when solving the merging problems.

**Comment by danilexn** [comment]
Thanks so much, Marvin! I will run more tests. And you're right about the mtime. Will commit more changes later :)

**Pull request merged**